proposals/low-level-struct-improvements.md

```diff
@@ -351,31 +364,46 @@ Span<int> ComplexScopedRefExample(scoped ref Span<int> span)
+The presence of `ref` fields means the rules around method arguments must match need to be updated as a `ref` parameter can now be stored as a field in a `ref struct` argument to the method. Previously the rule only had to consider another `ref struct` being stored as a field. The impact of this is discussed in [the compat considerations](#compat-considerations). 
+
+The presence of `scoped` allows us to refine the rule to reduce the friction the rule creates. The `scoped` modifier lets us remove arguments from consideration as they cannot be returned from the method.
 
-> For a method invocation if there is an argument for a `ref` or `scoped ref` parameter of a `ref struct` type where the argument has *safe-to-escape* E1 then no argument may contribute a narrower *safe-to-escape* than E1.
+> For a method invocation if there is an argument for a `ref`, `scoped ref` or `out` parameter of a `ref struct` type where the argument has *safe-to-escape* E1 then 
+> 1. No other `ref struct` argument may contribute a narrower *safe-to-escape* than E1.
+> 2. No other `ref` or `in` argument may contribute a narrower *ref-safe-to-escape* than E1.
 
 Impact of this change is discussed more deeply [below](#examples-method-arguments-must-match). Overall this will allow developers to make call sites more flexible by annotating non-escaping ref-like values with `scoped`.
 
+The `scoped` modifier on parameters also impacts our object overriding, interface implementation and `delegate` conversion rules. The signature for an override, interface implementation or `delegate` conversion can: 
+- Add `scoped` to a `ref` or `in` parameter
+- Add `scoped` to a `ref struct` parameter 
+Any other difference with respect to `scoped` will be considered an error. 
+
+The `scoped` modifier also has the following effects on method signatures:
+- The `scoped` modifier though does not affect hiding.
+- Overloads cannot differ only on `scoped`
+
 The section on `ref` field and `scoped` is long so wanted to close with a brief summary of the proposed breaking changes:
 
 * A value that has *ref-safe-to-escape* to the *calling method* is returnable by `ref` or `ref` field.
-* A `out` parameter would be considered `ref-safe-to-escape` within the *current method*.
+* A `out` parameter would be considered `safe-to-escape` within the *current method*.
 
 Detailed Notes:
 - A `ref` field can only be declared inside of a `ref struct` 
 - A `ref` field cannot be declared `static`, `volatile` or `const`
+- A `ref` field cannot have a type that is `ref struct`
 - The reference assembly generation process must preserve the presence of a `ref` field inside a `ref struct` 
 - A `readonly ref struct` must declare its `ref` fields as `readonly ref`
-- The span safety rules document will be updated as outlined in this document.
+- A `scoped` modifier must appear before `in`, `out`, or `ref`
```

Comment:
>must appear before [](http://example.com/codeflow?start=22&length=18)

"may appear before"?

Reply:
I prefer **must** here because it leaves open the door to do `ref scoped` in the future.

Reply:
"Must" could imply that all `in`, `out`, and `ref` modifiers require a preceding `scoped` modifier. Would it make sense to qualify the statement? Perhaps: "For by-ref values, the `scoped` modifier must appear before `in`, `out`, or `ref`"